tests/integration/test_charm.py

```diff
@@ -198,3 +201,123 @@ async def test_scale_down_capablities(ops_test: OpsTest) -> None:
     member_ips = await fetch_replica_set_members(ip_addresses, ops_test)
 
     assert set(member_ips) == set(ip_addresses), "mongod config contains deleted units"
+
+
+async def test_replication_across_members(ops_test: OpsTest) -> None:
+    """Check consistency, ie write to primary, read data from secondaries."""
+    # first find primary, write to primary, then read from each unit
+    ip_addresses = [unit.public_address for unit in ops_test.model.applications[APP_NAME].units]
+    primary = await replica_set_primary(ip_addresses, ops_test)
+    password = await get_password(ops_test)
+    client = MongoClient(unit_uri(primary, password), directConnection=True)
+
+    db = client["new-db"]
+    test_collection = db["test_collection"]
+    ubuntu = {"release_name": "Focal Fossa", "version": 20.04, "LTS": True}
+    test_collection.insert(ubuntu)
+
+    client.close()
+
+    secondaries = set(ip_addresses) - set([primary])
+    logger.error(secondaries)
```

Comment:
why logging secondaries with `logger.error`?
Also on L229. For visibility?

Reply:
heck so embarsed this made it into the PR 🤦